<?xml version="1.0" encoding="UTF-8"?>

<launch>
    <arg name="color_path" default="$(find yolov3-ros)/src/colors" />
    <arg name="weight_path" default="$(find yolov3-ros)/src/weights" />
    <arg name="config_path" default="$(find yolov3-ros)/src/cfg" />
    <arg name="class_path" default="$(find yolov3-ros)/src/data" />

    <arg name="weight_name" default="yolov3_ckpt_100.pth" />
    <arg name="config_name" default="yolov3.cfg" />
    <arg name="class_name" default="custom.names" />

    <node pkg="yolov3-ros" type="yolov3_ros.py" name="yolov3_node" output="screen">
        <param name="model_path" type="string" value="$(arg config_path)/$(arg config_name)" />
        <param name="weight_path" type="string" value="$(arg weight_path)/$(arg weight_name)" />
        <param name="class_path" type="string" value="$(arg class_path)/$(arg class_name)" />
        <param name="color_path" type="string" value="$(arg color_path)" />
        <remap from="image" to="/camera/rgb/image_rect_color/compressed"/>
    </node>
</launch>